vhost-device-console/src/backend.rs

```diff
@@ -73,23 +76,33 @@ impl VuConsoleConfig {
         (0..self.socket_count).map(make_socket_path).collect()
     }
 
-    pub fn generate_tcp_addrs(&self) -> Vec<String> {
-        let tcp_port_base = self.tcp_port.clone();
-
-        let make_tcp_port = |i: u32| -> String {
-            let port_num: u32 = tcp_port_base.clone().parse().unwrap();
-            "127.0.0.1:".to_owned() + &(port_num + i).to_string()
-        };
-
-        (0..self.socket_count).map(make_tcp_port).collect()
+    pub fn generate_vm_sock_addrs(&self) -> Vec<String> {
+        match self.backend {
+            // if type is Nested, result will be dropped.
+            BackendType::Nested => {
+                vec![String::new()]
+            }
+
+            BackendType::Network => {
+                let port_base: u32 = self.tcp_port.parse().unwrap();
+                let make_tcp_port =
+                    |i: u32| -> String { "127.0.0.1:".to_owned() + &(port_base + i).to_string() };
+                (0..self.socket_count).map(make_tcp_port).collect()
+            }
+
+            BackendType::Uds => {
+                let make_uds_path = |i: u32| -> String { self.vm_sock.to_owned() + &i.to_string() };
+                (0..self.socket_count).map(make_uds_path).collect()
+            }
+        }
     }
 }
 
 /// This is the public API through which an external program starts the
 /// vhost-device-console backend server.
 pub fn start_backend_server(
     socket: PathBuf,
-    tcp_addr: String,
+    vm_sock: String,
```

Comment:
IIUC `vm_sock` is used for any backend now, so it's another point to change the parameter name to something different, otherwise it's very confusing.

If we change the parameter to `uds_path`, I think `vm_sock` here is fine, otherwise please use a different name.